tests/queries/0_stateless/02985_variant_type_analytical.sh

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env bash
+# Tags: long
+
+CUR_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# reset --log_comment
+CLICKHOUSE_LOG_COMMENT=
+# shellcheck source=../shell_config.sh
+. "$CUR_DIR"/../shell_config.sh
+
+CH_CLIENT="$CLICKHOUSE_CLIENT --allow_experimental_variant_type=1 --use_variant_as_common_type=1 --allow_experimental_object_type=1"
+
+
+function create_table()
+{
+    $CH_CLIENT -q "create table test (id UInt64, v Variant(UInt64, String, Float64)) engine=$1;"
+}
+
+function drop_table()
+{
+    $CH_CLIENT -q "drop table if exists test;"
+}
+
+function test_order_by
+{
+    create_table "$1"
+
+    $CH_CLIENT -q  "insert into test values(1, 4);"
+    $CH_CLIENT -q  "insert into test values(2, 2);"
+    $CH_CLIENT -q  "insert into test values(3, 1);"
+    $CH_CLIENT -q  "insert into test values(4, 5);"
+    $CH_CLIENT -q  "insert into test values(5, 3);"
+
+    $CH_CLIENT -q  "insert into test values(6, 'b');"
+    $CH_CLIENT -q  "insert into test values(7, 'd');"
+    $CH_CLIENT -q  "insert into test values(8, 'a');"
+    $CH_CLIENT -q  "insert into test values(9, 'c');"
+    $CH_CLIENT -q  "insert into test values(10, 'e');"
+
+    $CH_CLIENT -q  "insert into test values(11, 12.0);"
+    $CH_CLIENT -q  "insert into test values(12, 13.5);"
+    $CH_CLIENT -q  "insert into test values(13, 15.0);"
+    $CH_CLIENT -q  "insert into test values(14, 13.0);"
+    $CH_CLIENT -q  "insert into test values(15, 15.20);"
+    $CH_CLIENT -q  "insert into test values(16, 11.0);"
+    $CH_CLIENT -q  "insert into test values(17, 15.25);"
+    $CH_CLIENT -q  "insert into test values(18, 14.0);"
+
+    #$CH_CLIENT -nmq "select id, v, variantType(v) from test order by v;" # Does not sort
```

Comment:
Variant type is not comparable (obviously because it ca contain types that cannot be compared bwtween each other). So, `order by` can return arbitrary order and it's expected. I will add it to the documentation

Reply:
Hm, maybe we can make it partially comparable, need to think about it

Reply:
Yes but it will be really hard when complex types are involed. Restricting using order by clause without specifying the data type is much cleaner than returning an arbitrary order IMO.